### ISSUE TITLE
English Search POI: Guy's != Guys (Mcdonalds)

### ISSUE THREAD

**Issue by pnoll1** [osmandapp/OsmAnd#4491]
What version of OsmAnd are you using, on what device and Android/iOS version?
OsmAnd 2.7.5
Do you use the offline maps offered within the OsmAnd app for download, or online (tile / raster) maps?
Offline maps

OSM POI results that include an apostrophe in the name will not be found if the apostrophe is omitted when searching in OsmAnd.

For Example:
Putting five guys into the search line will not show locations named five guy's in OSM.
Best illustrated by navigating to Lat 47.71121 Lon -122.18791, search for five guys, nearest will be ~4.6 miles.  Return to lat and long and search for five guy's, the nearest will be a few hundred feet away.

**Comment by pnoll1** [comment]
#4803 seems to be similar but deals with french usage of apostrophe.  The user expects search function to find an item with apostrophe in the name in OSM data but with apostrophe omitted when searching.

**Comment by vshcherb** [comment]
"Guy's" and "Guys" have different meaning & semantics, though they are phonetically the same but our search doesn't support phonetical symmetry.

**Comment by pnoll1** [comment]
I agree with the title change and @vshcherb comment.  This is more of feature request than a bug.  That said most users would expect some sort of search suggestion system.  I can find multiple other open issues from users  #4870, #4700 that all stem from the lack of this system.

**Comment by tbertels** [comment]
Couldn't apostrophes be considered as spaces? Same thing for hyphen/dash.
It should be easier to implement while there's no phonetic search library and would already improve searches a lot.

**Comment by tbertels** [comment]
Indeed, it needs to be more intelligent than that.
It should remove the ' (apostrophe) by default (e.g. for guy's), unless there's only one letter before the ' (e.g. in l'avion).
Applying this transformation to both the searched expression and the target is of course necessary.

**Comment by Marenz** [comment]
what about transformations like `southbeach` => `south beach`? Right now it wouldn't find anything for the former..

**Comment by Marenz** [comment]
Is there maybe an issue collecting all the search feature enhancement suggestions?

**Comment by Zirochkabila** [comment]
Example: "Rue helvétie" will never be found, because of the beug it is imperative to type "rue de l'helvetie".
OsmAnd~ 4.4.0#14149mqta, released: 2023-02-02

**Comment by tbertels** [comment]
Even "helvétie" won't be found, and users usually don't bother to enter "rue de l'" before it. This very important in French speaking countries.

So it seems like a regex is needed:
`Guy's` and `Guys` should both be found when searching for either `guy's`, `guys` or `guy`.

**Comment by BudgieInWA** [comment]
I encountered this issue today (OsmAnd 4.8.6), searching for "Butters place" looking for Butter's Place
https://osmand.net/map?pin=-31.94057,115.82967

From my perspective as a user, the idea that *"Guy's" and "Guys" have different meaning & semantics* is of no relevance to this address/location search feature. I agree that users expect search to be "fuzzy" enough to handle this case, especially from a phone app, where text input can be tricky.

This has become important with the introduction of AndroidAuto support. Text input is limited to speech-to-text through that interface, making it impossible to search for such places.

My example was actually through AndroidAuto: first by asking the Voice Assistance for "directions to butters place", then by opening the search interface in OsmAnd on the car's screen and using the microphone icon to dictate "butters place", neither of which worked.

**Comment by vshcherb** [comment]
1. + Small update "helvétie" - is searchable

<img width="353" alt="Screenshot 2025-01-04 at 22 52 36" src="https://github.com/user-attachments/assets/1050a4c9-3f2a-4048-a30d-e3fe56282828" />

2. +"Five guys" issue is gone cause now all OSM nodes are correctly marked without appostrophe
3. "Guy's" - supermarket https://www.openstreetmap.org/node/4333505925 searchable by "guy's",  but not as "guys" or "guy"

**Comment by xmd5a2** [comment]
Guy's is searchable only by "guy's"
![Image](https://github.com/user-attachments/assets/89d72ecc-0038-405c-86a4-22a0726c95e4)
![Image](https://github.com/user-attachments/assets/d4a0c1f5-999b-4ed3-aa4f-d3f97bbad038)
![Image](https://github.com/user-attachments/assets/29b5a9ba-7339-41f5-868a-2bdf6bb975ed)

**Comment by Nicthurne** [comment]
I just came across this issue when trying to search for a POI that is properly called "Purz'ls Paradiesgarten" ([https://www.openstreetmap.org/way/250559036](https://www.openstreetmap.org/way/250559036)).

I expected the search function to find this POI by simply typing "Purzls" instead of "Purzl's", given that this would be a convenience feature.
Especially on a standard mobile keyboard, the apostrophe is hidden away and needs to be accessed via extra clicks in order to type it. Making the search function easier to use by allowing people to not type the apostrophe and still find what they search for would be great.

**Comment by erik55** [comment]
how about a fuzzy search? As said, organicmaps/comaps does find all mcdonalds when i search like that. As does google of course. Many people don’t even know how to spell it correctly, you say it often, never write it. It is sad that you have to do a google search first (or wikipedia) to see how you spell it correctly. And then there is even the mc donald’s or mc donald's and mcdonald’s or mcdonald's. Impossible to try all variants if you are in a hurry and just want to find something.

**Comment by scaidermern** [comment]
> how about a fuzzy search?

For fuzzy search there is ticket #5409.

**Comment by vshcherb** [comment]
Fixed now `guys` and `guy's`, `mcdonalds` and `mcdonald's` are interchangeable

**Comment by Zirochkabila** [comment]
# Testing 1

> Web -prod
> OsmAnd~ 5.3.0#6572m, released: 2025-11-25
> iOS - **waiting for**

`guys` = `guy's` (Five) \ `Mcdonalds` = `Mcdonald's`


<details><summary>Details</summary>
<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/82637642-7df2-454c-984b-0268a7ca1adf"  width="800" height="auto"/> </th>
    <td><video source src="https://github.com/user-attachments/assets/7e9a6625-9138-4c18-9f08-508d5f58fd57"  width="300" height="auto"/></td>
  </tr>
</table>  
</details>

**Comment by erik55** [comment]
> Fixed now `guys` and `guy's`, `mcdonalds` and `mcdonald's` are interchangeable

I hope it also works with all kind of correct and incorrect apostrophes like

* `guy’s` (typographically correct)
* ``guy`s`` (incorrect: accent)
* `guy´s` (incorrect: accent)

Can somebody test this?

**Comment by Zirochkabila** [comment]
Coordinates of place `47.71143/-122.19004`

@erik55 It doesn't work with ‘incorrect’ apostrophes. But, in my opinion, this is a completely expected result (they are different symbols).

<details><summary>Details</summary>
<table style="width:100%">
  <tr>
 <th><img src="https://github.com/user-attachments/assets/ad508483-2cf0-405e-a573-f64b2c56a1e9"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/1abdf481-1c22-40ab-bd5d-9b2035499a19"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/895fca65-4e6b-4ee6-9514-e99b08e30871"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/bf78b284-7bcd-4123-a182-3498bf19dffd"  width="300" height="auto"/> </th>
  </tr>
</table>      
</details>

**Comment by erik55** [comment]
> Coordinates of place `47.71143/-122.19004`
> 
> [@erik55](https://github.com/erik55) It doesn't work with ‘incorrect’ apostrophes. But, in my opinion, this is a completely expected result (they are different symbols).

I think this **is indeed** a problem, as in all Spanish speaking countries and in German speaking countries the ` or the ´ (accents) are used a lot instead of the (typographic or replacement) apostrophe ’ or '. In total a huge amount of world population speaks/writes those two languages. And similar confusions might occur in other cultures.

See the following examples: Punctuation (green) and typewriter (red) apostrophe, followed by a [prime](https://en.wikipedia.org/wiki/Prime_(symbol)) (blue), between letters Í and í (using [acute accent](https://en.wikipedia.org/wiki/Acute_accent)), using the [fonts](https://en.wikipedia.org/wiki/Font): [Arial](https://en.wikipedia.org/wiki/Arial), [Calibri](https://en.wikipedia.org/wiki/Calibri), [Tahoma](https://en.wikipedia.org/wiki/Tahoma_(typeface)), [Times New Roman](https://en.wikipedia.org/wiki/Times_New_Roman), and [Linux Libertine](https://en.wikipedia.org/wiki/Linux_Libertine)

![](https://upload.wikimedia.org/wikipedia/commons/5/5a/Apostrophe%2Bprime%2Bacute-in-_Arial%2BCalibri%2BTahoma%2BTimes%2BLibertine.png)


See this common example of wrong usage (accent instead of apostrophe):

![](https://upload.wikimedia.org/wikipedia/commons/b/b8/Marburg_-_Weidenh%C3%A4user_Stra%C3%9Fe_06_ies.jpg)

**Comment by Zirochkabila** [comment]
@Gateway07 can we try to fix it too?

> See the following examples: Punctuation (green) and typewriter (red) apostrophe, followed by a [prime](https://en.wikipedia.org/wiki/Prime_(symbol)) (blue), between letters Í and í (using [acute accent](https://en.wikipedia.org/wiki/Acute_accent)), using the [fonts](https://en.wikipedia.org/wiki/Font): [Arial](https://en.wikipedia.org/wiki/Arial), [Calibri](https://en.wikipedia.org/wiki/Calibri), [Tahoma](https://en.wikipedia.org/wiki/Tahoma_(typeface)), [Times New Roman](https://en.wikipedia.org/wiki/Times_New_Roman), and [Linux Libertine](https://en.wikipedia.org/wiki/Linux_Libertine)

**Comment by vshcherb** [comment]
Yes, this is mostly prepared. We normalize text when we create maps and we search. We need to double check it works for all cases
- [x] Search https://github.com/osmandapp/OsmAnd/blob/master/OsmAnd-java/src/main/java/net/osmand/search/core/SearchPhrase.java#L108 
- [x] Search iOS 
- [x] IndexAddressCreator
- [x] IndexPoiCreator

**Note**: as Algorithms.normalizeSearchText parameter should be added to deal with braces or just apostrophes. It's better not to use CollatorStringMatcher.alignChars! As we can simply align chars in the map data (address / POI) and query

**Comment by ivanPyrohivskyi** [comment]
- [x] [Ocean Pacific's Esmeralda](https://www.openstreetmap.org/way/447103795), Ocean Pacific´s Esmeralda
- [x] O´Pulpiño, [O'Pulpiño](https://www.openstreetmap.org/node/10090035165)
- [x] Braces: [Instituto de Crédito Oficial (ICO)](https://www.openstreetmap.org/node/1474148137), [Agencia Para el Empleo (Zona de Arganzuela)](https://www.openstreetmap.org/way/115982358)

**Comment by erik55** [comment]
Great, but it seems that you forgot one of the two common accents used as apostrophes in Europe:

* **guy`s** (incorrect: accent)
* guy´s (incorrect: accent)

![](https://s3-media0.fl.yelpcdn.com/bphoto/ku4qk9WEaMkLIC_fif90Nw/o.jpg)

**Comment by Zirochkabila** [comment]
# Testing 2

> 5.3.0.5309
> OsmAnd~ 5.3.0#6692m, Released: 2025-12-15
> Web - prod

Search behaves differently depending on the exact apostrophe character used in the query - **Found:**

  * `Mandy's Frizer` (ASCII apostrophe `'` – U+0027)
  * `Mandy´s Frizer` (acute accent `´` – U+00B4)

<details><summary>Details - Mandy's Frizer</summary>
<table style="width:100%">
  <tr>
    <td><img src="https://github.com/user-attachments/assets/210907cc-f82e-4270-b126-b2d4f8aa1847" width="900"/></td>
    <td><img src="https://github.com/user-attachments/assets/1ec6b8a0-e1b9-4921-a696-f69040168589" width="300"/></td>
  </tr>
</table>
</details>

<details><summary>Details - Mandy´s Frizer</summary>
<table style="width:100%">
  <tr>
    <td><img src="https://github.com/user-attachments/assets/0df8f49e-c871-48a8-aafd-1b8f1afdaf07" width="900"/></td>
    <td><img src="https://github.com/user-attachments/assets/17f07720-b0de-429d-bb83-4b17b5e60ddc" width="300"/></td>
  </tr>
</table>
</details>


_______

### Severity: 🟡 Minor

> Priority: 🔵 Low / Frequency: Very low

* [ ] 1. Inconsistent handling of apostrophe-like characters in POI name search (`Mandy's Frizer`) - **Not found**

  * `Mandy′s Frizer` (prime `′` - U+2032)
  * `Mandy‵s Frizer` (reversed prime `‵` - U+2035)
  * `Mandyʹs Frizer` (modifier letter prime `ʹ` - U+02B9)

<details><summary>Details - Mandy′s Frizer</summary>
<table style="width:100%">
  <tr>
    <td><img src="https://github.com/user-attachments/assets/c26880d5-b863-46c9-9538-b050c4b3a00e" width="900"/></td>
    <td><img src="https://github.com/user-attachments/assets/0f0f6e4c-4b97-48b2-beec-324dcd44d224" width="300"/></td>
  </tr>
</table>
</details>

<details><summary>Details - Mandy‵s Frizer</summary>
<table style="width:100%">
  <tr>
    <td><img src="https://github.com/user-attachments/assets/7f99b528-5cd4-463f-b414-6364c1af8578" width="900"/></td>
    <td><img src="https://github.com/user-attachments/assets/baa4230f-d78c-4f27-8de1-af39aa70c253" width="300"/></td>
  </tr>
</table>
</details>

<details><summary>Details - Mandyʹs Frizer</summary>
<table style="width:100%">
  <tr>
    <td><img src="https://github.com/user-attachments/assets/b484ffc7-c37d-4b5e-9243-57042c7481d5" width="900"/></td>
    <td><img src="https://github.com/user-attachments/assets/d5262545-7105-497e-8dea-11ebfc1f471e" width="300"/></td>
  </tr>
</table>
</details>

**Comment by erik55** [comment]
> # Testing 2
> 
> Details - Mandy's Frizer
> 	
> Details - Mandy´s Frizer
> 
> Details - Mandy′s Frizer
> 	
> Details - Mandy‵s Frizer
> 	
> Details - Mandyʹs Frizer

You forgot the grave accent, used in the above photo and is easy to type on European keyboards ⇒ therefore widely used as apostrophe.

**Comment by Zirochkabila** [comment]
# Testing 3

> OsmAnd Maps 5317, released 12.01.26
> OsmAnd~ 5.3.0#6892m, released: 2026-01-12

`52.549579/13.458186`

### **Severity**: 🟠 Major / **Priority**: 🟠 Normal / **Frequency**: Medium

- [ ] 1. Inconsistent handling of apostrophe-like characters in POI name search (`Mandy's Frizer`) - **Web only** still present

* `Mandy′s Frizer` (prime `′` — U+2032)
* `Mandy‵s Frizer` (reversed prime `‵` — U+2035)
* `Mandyʹs Frizer` (modifier letter prime `ʹ` — U+02B9)


<details><summary>Details - Mandy′s Frizer</summary>
<table style="width:100%">
  <tr>
    <td><img src="https://github.com/user-attachments/assets/ac686da8-4b01-42ab-84b4-7b58eb6ad881" width="900"/></td>
    <td><img src="https://github.com/user-attachments/assets/0aa890dc-37ba-4707-9dff-e6cd86a8c525" width="300"/></td>
  </tr>
</table>
</details>

<details><summary>Details - Mandy‵s Frizer</summary>
<table style="width:100%">
  <tr>
    <td><img src="https://github.com/user-attachments/assets/4e8a5745-bcd6-4ada-a39c-88e50b11b2d2" width="900"/></td>
    <td><img src="https://github.com/user-attachments/assets/11c5fa6d-ddd0-42cc-8719-ff1a8e9424f2" width="300"/></td>
  </tr>
</table>
</details>

<details><summary>Details - Mandyʹs Frizer</summary>
<table style="width:100%">
  <tr>
    <td><img src="https://github.com/user-attachments/assets/9450452b-dbfa-4597-b353-f95929a776ad" width="900"/></td>
    <td><img src="https://github.com/user-attachments/assets/6c05c7d5-4fb9-4afd-9380-34f0e9481719" width="300"/></td>
  </tr>
</table>
</details>

**Comment by vshcherb** [comment]
Updated web server

**Comment by Zirochkabila** [comment]
### Retest

> Web - prod

**Fixed**

<details><summary>Details</summary>
<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/dc892f1c-463f-4d28-ad19-86cb1b51ba04"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/b0552a59-e7db-4764-9710-facd088801ab"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/c1965b6d-fdde-4f07-abe7-6d32b614f922"  width="300" height="auto"/> </th>
  </tr>
</table>  
</details>
